lotemplate/classes.py

```diff
@@ -321,6 +344,100 @@ def fill(self, variables: dict[str, dict[str, Union[str, list[str]]]]) -> None:
         :return: None
         """
 
+        def ifReplace(doc, variables: dict[str, dict[str, Union[str, list[str]]]]) -> None:
+            """
+            Parse statements like [if $myvar==TOTO]...[endif]
+
+            If the condition matches we remove the if and endif statement.
+            It the condition doesn't match, we remove the statements and the text between the statements.
+
+            :param doc: the document to fill
+            :param variables: the variables
+            :return: None
+            """
+            def displayIfError(xFound, ifString, message):
```

Comment:
Very interesting, however the fact of displaying the errors in the document at filling rather than raising an error at scanning seems unnatural to me. It looks promising though.